modules/battery_manager/php/battery_manager.class.inc

```diff
@@ -0,0 +1,249 @@
+                new \LORIS\Data\Filters\UserSiteMatch()
+            );
+        }
+        return $provisioner;
+    }
+
+    function getOptions() {
+        $options = [
+                'instruments' => \Utility::getAllInstruments(),
+                'stages'      => $this->getStageList(),
+                'subprojects' => \Utility::getSubprojectList(null),
+                'visits'      => \Utility::getVisitList(),
+                'sites'       => \Utility::getSiteList(false),
+                'firstVisits' => $this->getYesNoList(),
+                'active'      => $this->getYesNoList(),
+               ];
+        return json_encode($options);
+    }
+
+    /**
+     * Return associative array of stages
+     *
+     * @return array
+     */
+    function getStageList() : array
+    {
+        return array(
+                "Not Started"   => 'Not Started',
+                "Screening"     => 'Screening',
+                "Visit"         => 'Visit',
+                "Approval"      => "Approval",
+                "Subject"       => "Subject",
+                "Recycling Bin" => "Recycling Bin",
+               );
+    }
+
+    /**
+     * Return associative array of yes and no values
+     *
+     * @return array
+     */
+    function getYesNoList() : array
+    {
+        return array(
+                'Y' => 'Yes',
+                'N' => 'No',
+               );
+    }
+
+    function toggleStatus($db, $id, $status) {
+        try {
+          $db->update('test_battery', array('Active' => $status), array('ID' => $id));
+          return $this->response(200, 'Availability Updated Successfully');
+        } catch (\DatabaseException $e) {
+          return $this->response(400, 'Could not update entry in the Test Battery');
+        }
+    }
+
+    function addTest($db, $user, $data) {
+        if (!$user->hasPermission('battery_manager_edit')) {
+            return $this->response(400, 'Edit Permission Denied');
+        }
+
+        if (!$this->isDuplicate($db, $data)) {
+            return $this->response(400, 'This Test already exists in the database');
+        }
+
+        $test = array(
+                'Test_name'    => $data['testName'],
+                'AgeMinDays'   => $data['ageMinDays'],
+                'AgeMaxDays'   => $data['ageMaxDays'],
+                'Stage'        => $data['stage'],
+                'SubprojectID' => $data['subproject'],
+                'Visit_label'  => $data['visitLabel'],
+                'CenterID'     => $data['centerId'],
+                'firstVisit'   => $data['firstVisit'],
+                'instr_order'  => $data['instrumentOrder'],
+                'Active'       => 'Y',
+        );
+
+        try {
+          $db->insert('test_battery', $test);
+          return $this->response(200, 'Test Submission Successful');
+        } catch (\DatabaseException $e) {
+          return $this->response(400, 'Could not add entry to the Test Battery');
```

Comment:
If there was a database exception that's a 500 server error, not a 400 client error.